src/App/Client.cpp

```
@@ -63,11 +65,18 @@ void sedp::Client::initialise_fields(const string& filename)
   gf2n::init_field(128); // Assumes 128-bit prime generation
 }
 
-int sedp::Client::connect_to_player(string ip, int port) {
+SSL * sedp::Client::connect_to_player(string ip, int port) {
 
   int socket_id = OpenConnection(ip, port);
-
-  return socket_id;
+  SSL_CTX * ctx;
+  LoadCertificates(ctx, "/home/gpik/SCALE-MAMBA/Cert-Store/RootCA.crt","/home/gpik/SCALE-MAMBA/Cert-Store/RootCA.key");
+  ssl = SSL_new(ctx);
+  SSL_set_fd(ssl, socket_id);
+  if ( SSL_connect(ssl) <= 0 ){   /* perform the connection */
+    ERR_print_errors_fp(stderr);
```

You should throw an error also. See what kind of errors SCALE supports at `Exceptions.h`